src/main/java/org/openrewrite/java/migrate/DetectAWTGetPeerMethod.java

```diff
@@ -0,0 +1,117 @@
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.ChangeMethodName;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.search.UsesMethod;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+@AllArgsConstructor(access = AccessLevel.PACKAGE) // For tests
+class DetectAWTGetPeerMethod extends Recipe {
+    String methodPatternGetPeer;
+    String methodUpdateIsDisplayable;
+    String className;
+    String methodUpdateIsLightweight;
+
+    @Override
+    public String getDisplayName() {
+        return "Replace `getPeer()` method";
+    }
+
+    @Override
+    public String getDescription() {
+        return " All methods that refer to types defined in the java.awt.peer package are removed in Java 11. "
+               + "This recipe replaces the use of getPeer() method in the java.awt.Component, java.awt.Font, and java.awt.MenuComponent classes and direct known subclasse."
+               + "The occurrence of  `(component.getPeer() != null) { .. }` is replaced with `(component.isDisplayable())` "
+               + "and the occurrence of `(component.getPeer() instanceof LightweightPeer)` "
+               + "is replaced with `(component.isLightweight())`.";
+    }
+
+    @JsonCreator
+    public DetectAWTGetPeerMethod() {
+        this.methodPatternGetPeer = "java.awt.* getPeer()";
+        this.methodUpdateIsDisplayable = "java.awt.* isDisplayable()";
+        this.className = "java.awt.peer.LightweightPeer";
+        this.methodUpdateIsLightweight = "java.awt.*  isLightweight()";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(new UsesMethod<>(methodPatternGetPeer), new JavaVisitor<ExecutionContext>() {
+            @Override
+            public <T extends J> J.ControlParentheses<T> visitControlParentheses(J.ControlParentheses<T> cp, ExecutionContext ctx) {
+                Expression ifCExp = (Expression) cp.getTree();
+                //Check if there is a Binary or an instanceOf inside the paranthesis
+                if (!(ifCExp instanceof J.Binary) && !(ifCExp instanceof J.InstanceOf)) {
+                    return cp;
+                }
+                if (ifCExp instanceof J.Binary) {
+                    J.Binary binaryCondition = (J.Binary) ifCExp;
+                    //check if(x.getPeer() != null)
+                    if (binaryCondition.getLeft() instanceof J.MethodInvocation && binaryCondition.getRight() instanceof J.Literal && binaryCondition.getOperator().name().equals("NotEqual")) {
+                        J.MethodInvocation mi = (J.MethodInvocation) binaryCondition.getLeft();
+                        J.Literal lt = (J.Literal) binaryCondition.getRight();
+                        if (mi.getName().getSimpleName().equals("getPeer") && lt.getValueSource().equals("null")) {
```

Comment:
Instead of comparing method name, let's use a type safe MethodMatcher using `methodPatternGetPeer`.